webapp/app.py

```diff
@@ -215,6 +216,39 @@ def start_career():
     return flask.render_template("/careers/start.html")
 
 
+@app.route("/careers", methods=["GET"])
+def all_careers():
+    context = {
+        "all_departments": _group_by_department(greenhouse.get_vacancies())
+    }
+    context["vacancies"] = greenhouse.get_vacancies()
+    context["vacancies_json"] = [
+        vacancy.to_dict() for vacancy in context["vacancies"]
+    ]
+
+    return flask.render_template("/careers/index.html", **context)
+
+
+@app.route("/careers/all", methods=["GET"])
+def all_jobs():
+    context = {
+        "all_departments": _group_by_department(greenhouse.get_vacancies())
+    }
+    context["vacancies"] = greenhouse.get_vacancies()
```

Comment:
So, we simply pass through all vacancies to the page.

What, then, is the point of only showing the first 20 and having a "show 20 more" button? Normally I would assume this is so we don't have to send all the data initially, presumably reducing the size of the payload so the initial page loads quicker, but this isn't the case here. Which means there are two further possible benefits:

- Maybe even though we're sending all the data, the page *renders* slower if we display all the roles right away. I'm sceptical about this, I don't think it's self-evident. But if this is part of the rationale here I'm happy to put it to the test.
- Maybe we think that the "show 20 more" button is a better user experience? Again, I'm sceptical. I think it's surely most convenient for the user to simply scroll down the page to see more roles (we might consider making the filter list on the left sticky), plus they can then easily use CTRL+f to search the list without having to click the button a bunch of times first. But again if I've missed some UX arguments here I'm happy to hear them.

So if I'm right about this, then the question becomes: Is it too expensive to send all roles in the initial pageload? should we only send 20 in the first load and lazy-load more of them after the fact? Or is it in fact fast enough with sending all roles in the first load that we could simply display the full list right at the start?

From playing with the demo, there doesn't seem to be any significant delay on loading that page really. It seems surprisingly quick. So I wonder if we might want to just do away with the "show 20 more" stuff completely and just display the whole list?